Bottom: 89a1bd159135ab5aae205202e62cdc91b9ab3b0d
Top:    8d2f75897b59b2230d826c1aa131f191ac2a54cb
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-12-18 14:58:18 -0800

Refresh of dsocket

---

diff --git a/src/rsocket.c b/src/rsocket.c
index 219aa4a..42a28d2 100644
--- a/src/rsocket.c
+++ b/src/rsocket.c
@@ -2833,6 +2833,7 @@ static int rs_poll_arm(struct pollfd *rfds, struct pollfd *fds, nfds_t nfds)
 				else
 					rfds[i].fd = rs->cm_id->channel->fd;
 			} else {
+				printf("%s ready to poll epfd\n", __func__);
 				rfds[i].fd = rs->epfd;
 			}
 			rfds[i].events = POLLIN;
@@ -2841,7 +2842,6 @@ static int rs_poll_arm(struct pollfd *rfds, struct pollfd *fds, nfds_t nfds)
 			rfds[i].events = fds[i].events;
 		}
 		rfds[i].revents = 0;
-
 	}
 	return 0;
 }
@@ -2906,7 +2906,7 @@ int rpoll(struct pollfd *fds, nfds_t nfds, int timeout)
 		if (ret)
 			break;
 
-		ret = poll(rfds, nfds, timeout);
+		ret = poll(rfds, nfds, timeout ? 10000 : 0);
 		if (ret <= 0)
 			break;
